src/app/WeaponPicker.tsx

```diff
@@ -0,0 +1,65 @@
+import { Autocomplete, Box, TextField, Typography } from "@mui/material";
+import { memo, useCallback, useMemo } from "react";
+import type { Weapon } from "../calculator/weapon";
+import { weaponTypeLabels } from "./uiUtils";
+
+export type WeaponOption = {
+  label: string; // weaponName
+  value: string; // weaponName
+  type: string; // weaponType
+};
+
+interface Props {
+  selectedWeapons: WeaponOption[];
+  onSelectedWeaponsChanged(weapons: WeaponOption[]): void;
+  weaponOptions: WeaponOption[];
+}
+
+const sortByTypeThenName = (a: Weapon, b: Weapon): number => {
+  if (a.weaponType < b.weaponType) return -1;
+  if (a.weaponType > b.weaponType) return 1;
+  // weapon types are the same, so compare the name
+  if (a.weaponName < b.weaponName) return -1;
+  if (a.weaponName > b.weaponName) return 1;
+  // both primary and secondary values are equal
+  return 0;
+};
+const makeOption = (weapon: Weapon): WeaponOption => ({
+  label: weapon.weaponName,
+  value: weapon.weaponName,
+  type: weaponTypeLabels.get(weapon.weaponType) || "",
+});
+
+export const makeWeaponOptionsFromWeapon = (weapons: Weapon[]): WeaponOption[] =>
+  weapons.sort(sortByTypeThenName).map(makeOption);
+
+/**
+ * An Autocomplete to allow for manually specifying weapons
+ */
+function WeaponPicker({ onSelectedWeaponsChanged, weaponOptions, selectedWeapons }: Props) {
+  const handleOnChange = useCallback(
+    (event: React.SyntheticEvent<Element, Event>, newSelection: WeaponOption[]) => {
+      onSelectedWeaponsChanged(newSelection);
+    },
+    [onSelectedWeaponsChanged],
+  );
+
+  return (
+    <Box>
+      <Typography component="h2" variant="h6" sx={{ mb: 1 }}>
+        Weapons
```

Comment:
I think this "weapons" title is a little redundant, because the input already says "weapons"

![image](https://github.com/user-attachments/assets/2dc9879e-0466-4fc4-b972-0a7b523ca6fd)

Maybe we remove this title, and place the search box above affinities so it doesn't look like it's part of the affinities section. What do you think?

Reply:
Sure! I'm on board.

<img width="305" alt="image" src="https://github.com/user-attachments/assets/3ebf1c04-3088-4445-a838-f2e766b54c2b">